### PR TITLE
Issue-842: Fix deprecation warnings during build time (external dependencies)

### LINKED ISSUE #842: Fix deprecation warnings during build time
There are some deprecation warnings when compiling the code. These need to be addressed.

Examples:
```
~/strongbox/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/MockedRemoteRepositoriesHeartbeatConfig.java
~/strongbox/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/StorageApiTestConfig.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/config/MavenIndexerConfig.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/MavenMetadataManager.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/metadata/MavenSnapshotManager.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/repository/group/metadata/MavenMetadataGroupRepositoryComponent.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/RepositoryIndexerFactory.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/RepositoryIndexer.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/downloader/IndexDownloader.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/IndexedMavenArtifactDeletedEventListener.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/artifact/MavenRepositoryArtifact.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactDeletedEventListener.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/BaseLocalStorageProxyRepositoryExpiredArtifactsCleanerTest.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/RetryDownloadArtifactTestBase.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/validation/groupid/MavenGroupIdLowercaseValidatorTest.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/config/Maven2LayoutProviderTestConfig.java
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/validation/artifactid/MavenArtifactIdLowercaseValidatorTest.java
Warning:(11, 26) java: org.mockito.Matchers in org.mockito has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/WhenRepositoryIsNotAliveDontCleanExpiredArtifactsTestIT.java
Warning:(18, 26) java: org.mockito.Matchers in org.mockito has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/testing/MavenTestCaseWithArtifactGeneration.java
Warning:(468, 56) java: toFile() in org.carlspring.strongbox.providers.io.RepositoryPath has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/WhenRepositoryIsAliveCleanExpiredArtifactsTestIT.java
Warning:(47, 35) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
Warning:(48, 34) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/CleanupExpiredArtifactsFromProxyRepositoriesCronJobTestIT.java
Warning:(130, 47) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
Warning:(131, 46) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
Warning:(138, 37) java: propagate(java.lang.Throwable) in com.google.common.base.Throwables has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java
Warning:(242, 43) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(279, 43) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(335, 43) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(357, 43) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/search/MavenIndexerSearchProviderTest.java
Warning:(164, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(180, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(195, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(211, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(227, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(243, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(258, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(274, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenProxyRepositoryProviderTestIT.java
Warning:(129, 34) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
Warning:(155, 34) java: containsPath(org.carlspring.strongbox.providers.io.RepositoryPath) in org.carlspring.strongbox.providers.layout.LayoutProvider has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/indexing/StrongboxIndexerTest.java
Warning:(111, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(132, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(211, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
Warning:(232, 34) java: validateAndStore(java.lang.String,java.lang.String,java.lang.String,java.io.InputStream) in org.carlspring.strongbox.services.ArtifactManagementService has been deprecated
~/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/ClearTrashCronJobFromMaven2RepositoryTestIT.java
Warning:(166, 29) java: propagate(java.lang.Throwable) in com.google.common.base.Throwables has been deprecated
Warning:(228, 29) java: propagate(java.lang.Throwable) in com.google.common.base.Throwables has been deprecated
Warning:(246, 122) java: toFile() in org.carlspring.strongbox.providers.io.RepositoryPath has been deprecated
Warning:(256, 122) java: toFile() in org.carlspring.strongbox.providers.io.RepositoryPath has been deprecated
```

### PR DESCRIPTION
**Pull request by v3rtumnus**
Fixes #842 

This is only the beginning but I also wanted to get feedback from your CI server.

### CODE REVIEW

**Comment by v3rtumnus** [comment]
@carlspring This pull now contains fixes to all deprecations in regards to external libraries, with one exception: The use of `StandardPasswordEncoder` in the class `CacheManagerAuthenticationCache`, since I was not sure if you really want to replace it with one of the other encoders. I will add a list of internal deprecations in issue #842

**Comment by carlspring** [comment]
Thanks for looking into it! So far, I think it looks good!

@sbespalov / @fuss86 : Any thoughts regarding the `StandardPasswordEncoder` and `CacheManagerAuthenticationCache` question?

@v3rtumnus : The list would be quite helpful, thanks!

**Comment by carlspring** [comment]
@v3rtumnus : Is this ready for merging?

**Comment by v3rtumnus** [comment]
It currently contains all fixes for external deprecation warnings (except the `StandardPasswordEncoder)`. If you want to do the internal ones (hope to provide the list in one of the next days) in a separate PR, then it is ready from my side.

**Comment by v3rtumnus** [comment]
Here is a list of internal deprecations:

```
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/validation/resource/ArtifactOperationsValidator.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/MockedRemoteRepositoriesHeartbeatConfig.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/RepositoryIndexer.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/storage/validation/groupid/MavenGroupIdLowercaseValidatorTest.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NugetLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/config/NugetLayoutProviderTestConfig.java: Some input files use or override a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NpmLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/NpmLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/RawLayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/RawLayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-raw-layout-provider/src/test/java/org/carlspring/strongbox/providers/layout/RawLayoutProviderTest.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-p2-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/P2LayoutProvider.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-p2-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/P2LayoutProvider.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/main/java/org/carlspring/strongbox/web/HeaderMappingFilter.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/main/java/org/carlspring/strongbox/web/HeaderMappingFilter.java uses or overrides a deprecated API.
[INFO] /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/test/java/org/carlspring/strongbox/config/IntegrationTest.java: /home/michael/Entwicklung/workspace_test/strongbox/strongbox-web-core/src/test/java/org/carlspring/strongbox/config/IntegrationTest.java uses or overrides a deprecated API.
```
Please let me know if you want it in a different format and what if I should add something to this PR.

**Comment by v3rtumnus** [comment]
Can be merged from my side, as the internal deprecations will be dealt with in a separate PR

File: strongbox-data-service/src/main/java/org/carlspring/strongbox/data/server/InMemoryOrientDbServer.java [hunk 1]
```diff
@@ -62,20 +62,6 @@ public void onDrop(ODatabaseInternal iDatabase)
         iDatabase.registerHook(genericEntityHook);
     }
 
-    @Override
-    public void onCreateClass(ODatabaseInternal iDatabase,
-                              OClass iClass)
-    {
-        iDatabase.registerHook(genericEntityHook);
```
Comment (fuss86) [inline strongbox-data-service/src/main/java/org/carlspring/strongbox/data/server/InMemoryOrientDbServer.java]: @sbespalov what do you think about this change ?

Comment (sbespalov) [reply strongbox-data-service/src/main/java/org/carlspring/strongbox/data/server/InMemoryOrientDbServer.java]: seems that this methods was deprecated
anyway I think this shouldn't affect for us, so let it be so

File: strongbox-commons/src/main/java/org/carlspring/strongbox/util/ClassLoaderFactory.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public static URLClassLoader urlClassLoaderFromPaths(ClassLoader parent,
         }
         catch (IOException e)
         {
-            throw Throwables.propagate(e);
+            throw new RuntimeException(e);
```
Comment (sbespalov) [inline strongbox-commons/src/main/java/org/carlspring/strongbox/util/ClassLoaderFactory.java]: I think that better to use `UndeclaredThrowableException` here and in similar cases

**Comment by v3rtumnus** [comment]
I incorporated the feedback by @sbespalov regarding exception handling and added one missed deprecation handling for mockito. Can by merged imho when the Jenkins server is back up.

**Comment by carlspring** [comment]
Thanks for the reviews and fixes! :)

**Comment by steve-todorov** [comment]
@carlspring I believe this can be merged now. Failing test cases are related to another issue. 
@v3rtumnus thanks for your contribution! :+1:  :)

**Comment by carlspring** [comment]
Oh, i didn't see the Sonar warnings before merging. @v3rtumnus : Would you mind cleaning up the warnings about the unused import statements and opening a new pull for this? Thanks!

**Pull request merged**